### DISCUSSION TITLE
How to add Action buttons?

### DISCUSSION THREAD

**Discussion by vahnmarty** [rappasoft/laravel-livewire-tables#635] [Q&A] [answered]
I want to add Edit, Delete and View button in every Row (grouped in a single cell).

**Comment by rappasoft** [comment, marked as answer]
```
Column::make('Actions')
    ->format(function($value, $column, $row) {
        return view('admin.user.actions')->withUser($row);
    }),
```

**Reply by kasirye** [reply]
Thanks for the feedback. how can i do the same on modals.

**Reply by lutfifirdaus** [reply]
maybe i can help answering too. You can create modal as one file blade and pass file path inside callback that rappasoft answer, or you can use modalViews(). See [here](https://rappasoft.com/docs/laravel-livewire-tables/v1/display/working-with-modals), hope this help you.

**Reply by jonathanoeijoeng** [reply]
I got error using this : Column not found: 1054 Unknown column 'master_products.actions' in 'field list'. 
Anything change since v2?

**Reply by jurri** [reply]
`format()` doesn't work, because the columns doesn't exist in datatable/model. 
For undefined column-names use `label()`.

return with view
```
Column::make('Actions')
    ->label(
        function ($row, Column $column) {
            return view('path.view');
        }
    ),
``` 

return with buttons
```
Column::make('Actions')
    ->label(
        function ($row, Column $column) {
            $delete = '<button class="bg-blue-500 hover:bg-blue-700 text-white font-bold p-2 rounded m-1" wire:click="delete(' . $row->id . ')">Trash</button>';
            $edit = '<button class="bg-blue-500 hover:bg-blue-700 text-white font-bold p-2 rounded m-1" wire:click="edit(' . $row->id . ')">Edit</button>';
            return $edit . $delete;
        }
    )->html(),
```

**Comment by gldtn** [comment]
Can you give an example on how the "admin.user.actions" would be setup please?

**Reply by lutfifirdaus** [reply]
you can create a linked button and pass the data as route parameter like this example

in component

```
public function columns(): array
    {
        return [
            Column::make("Name", "name")
                ->sortable()
                ->searchable(),
            Column::make("Email", "email")
                ->sortable()
                ->searchable(),
            Column::make('Action')
                ->format(function ($value, $column, $row) {
                    return view('admin.user.action', ['user' => $row]);
                }),
        ];
    }
```

in blade
```
<div class="d-flex justify-content-start">
    <a class="btn btn-sm btn-warning" href="{{ route('user.edit', $user) }}"><i class="fas fa-edit></i></a>

    {{-- you can add more button to what action you need --}}
</div>
```

it's make action row that inside there, you can add how much button you need, like my example its return edit user button that goes to edit page. You also can add a model too instead just button.

**Reply by jonathanoeijoeng** [reply]
does it mean we need to create 1 blade file which contain only button?

**Reply by lutfifirdaus** [reply]
> does it mean we need to create 1 blade file which contain only button?

yes you can do that, or if you want, you can write html syntax right inside that `format()` function and add `asHtml()` function after.

**Comment by Nemachtilli** [comment]
`Column::make("Actions")
                ->label(
                    fn($row, Column $column) => view('path.view')->with(['user' =>$row])
                ),`

**Comment by Nemachtilli** [comment]
`Column::make('Action')
            ->label(function ($row, Column $column) {
                    return view('path.view', ['user' => $row]);
                },
            ),`

**Comment by bhagat-abhishek** [comment]
i am having an issue, and it is that when i exclude the 'id' Column and get the row->id it returns the id in the row count instead of the post->id, how can i use it.

**Reply by lrljoe** [reply]
Use the setAdditionalSelects method to add ID as an additional select.  

Please do raise a new discussion rather than replying to a year old one that is marked as solved in future, as you may not get responses!

**Comment by bhagat-abhishek** [comment]
Thank you for your response. It worked for me.
